lib/src/strings.dart

```diff
@@ -195,22 +172,27 @@ String? _tryYamlEncodeFolded(String string, int indentSize, String lineEnding) {
 ///   my literal
 ///   string
 ///
-/// # With the "keep" chomping indicator
-/// key: |+
+/// # Without chomping indicator
+/// key: |
 ///   my literal
 ///   string
 /// ```
 ///
 /// See: https://yaml.org/spec/1.2.2/#812-literal-style
 String? _tryYamlEncodeLiteral(
     String string, int indentSize, String lineEnding) {
-  if (_shouldDoubleQuote(string)) return null;
+  if (_hasUnprintableCharacters(string)) return null;
+  // A string that starts with space or newline followed by space can't be
+  // encoded in literal mode.
+  if (string.startsWith(' ') && string.startsWith('\n ')) return null;
+
+  // TODO: Are there other strings we can't encode in literal mode?
 
   final indent = ' ' * indentSize;
 
   /// Simplest block style.
   /// * https://yaml.org/spec/1.2.2/#812-literal-style
-  return '|${_getChompingIndicator(string)}\n$indent'
+  return '|${string.endsWith('\n') ? '' : '-'}\n$indent'
       '${string.replaceAll('\n', lineEnding + indent)}';
 }
 
```

Comment:
@jonasfj I finally got around to look into this. Why the change? Literal strings are also subject to chomping as indicated [here](https://yaml.org/spec/1.2.2/#rule-b-chomped-last:~:text=Inside%20literal%20scalars%2C%20all%20(indented)%20characters%20are%20considered%20to%20be%20content%2C%20including%20white%20space%20characters.%20Note%20that%20all%20line%20break%20characters%20are%20normalized.%20In%20addition%2C%20empty%20lines%20are%20not%20folded%2C%20though%20final%20line%20breaks%20and%20trailing%20empty%20lines%20are%20chomped.)

This causes anything with more than one trailing `\n` to fail as every new line will be clipped such that:
1. `foo bar \n` remain `foo bar \n` (okay) 
2. `foo bar \n\n` becomes `foo bar \n` (not okay). 

That's why I went for a wildcard approach. We chomp so that it's always parsed correctly by `package: yaml` even if we only have a single line break.